ProgramHierarchy.java

```diff
@@ -1,18 +1,65 @@
-/*
- * File: ProgramHierarchy.java
- * Name: 
- * Section Leader: 
- * ---------------------------
- * This file is the starter file for the ProgramHierarchy problem.
- */
-
-import acm.graphics.*;
-import acm.program.*;
-import java.awt.*;
-
-public class ProgramHierarchy extends GraphicsProgram {	
-	public void run() {
-		/* You fill this in. */
-	}
-}
-
+/*
+ * File: ProgramHierarchy.java
+ * Name: 
+ * Section Leader: 
+ * ---------------------------
+ * This file is the starter file for the ProgramHierarchy problem.
+ */
+
+import acm.graphics.*;
+import acm.program.*;
+import java.awt.*;
+
+public class ProgramHierarchy extends GraphicsProgram {	
+	
+	
+	private static final int HEIGHT = 50;
+	private static final int WIDTH = 150;
+	
+	public void run() {
+		int x;
+		int y;
+		
+		x = (getWidth() - WIDTH) / 2;
+		y = (getHeight() - HEIGHT) /2;
+		
+		GRect Rect = new GRect(x, y, WIDTH, HEIGHT);
+		add(Rect);
+		
+		GLabel label1 = new GLabel("Program", (x + 50), (y + 30));
+		add(label1);
+		
+		
+		GLine Line1 = new GLine((x + 75), (y+50), (x+75), (y+100));
+		add(Line1);
+		
+		GRect Rect2 = new GRect(x, (y+100), WIDTH, HEIGHT);
+		add(Rect2);
+		
+		GLabel label2 = new GLabel("ConsoleProgram", (x + 30), (y + 130));
+		add(label2);
+		
+		GLine Line2 = new GLine((x - 100), (y + 100), (x + 75), (y + 50)); 
+		add(Line2);
+		
+		GRect Rect3 = new GRect((x - 175), (y + 100), WIDTH, HEIGHT);
+		add(Rect3);
+		
+		GLabel label3 = new GLabel("GraphicProgram", (x - 145), (y + 130));
+		add(label3);
+		
+		GLine Line3 = new GLine((x + 75), (y+50), (x+250), (y+100));
+		add(Line3);
+		
+		GRect Rect4 = new GRect((x+175), (y+100), WIDTH, HEIGHT);
+		add(Rect4);
+		
+		GLabel label4 = new GLabel("DialogProgram", (x + 210), (y + 130));

```

Comment:
Generally you should avoid the use of constant numbers in your program. You have values like (210, 130, 175, 145...etc) throughout your code and this is not a good idea because:
1. Someone reading your code will be confused how you came up with them
2. If requirements change your program will be really hard to update. For example imagine for this exercise the requirements changed and we decide that the value for HEIGHT (50) and WIDTH (150) should be changed to HEIGHT = 60 and 
WIDTH = 180

If this happens your program will fail to draw the right diagram. To make it work you will have to go back and start changing all the numbers throughout your program. It should be possible to come up with a solution that works without you doing further modification. The ability of a program to be able to adapt to changes like that is called scalability. So in this case your solution is not scalable enough.